src/query/execute_statement.c

```diff
@@ -13217,22 +13213,6 @@ do_prepare_insert (PARSER_CONTEXT * parser, PT_NODE * statement)
   class_ = statement->info.insert.spec->info.spec.flat_entity_list;
   values = statement->info.insert.value_clauses;
 
-  /* prepare only when simple insert clause are used */
-  if (values->info.node_list.list_type != PT_IS_VALUE)
-    {
-      return NO_ERROR;
-    }
-
-  /* prevent multi statements */
-  /* prevent multi values insert */
-  /* prevent do replace */
-  /* prevent dup key update */
-  if (pt_length_of_list (statement) > 1 || pt_length_of_list (values) > 1 || statement->info.insert.do_replace
-      || statement->info.insert.odku_assignments != NULL)
-    {
-      return NO_ERROR;
-    }
-
```

Comment:
@lucianlazar2211 

It may be better to give up multiple statements case, because the following logic assumes "one insert statement" is given. Since the case is very impractical, I don't mind leaving it as client insert.